app/data/views.py

```diff
@@ -331,57 +332,152 @@ def _query_param_to_case_stmnt(self, param, request, queryset):
+                        for week in easy_ranges['week_of_year']['range']
+                        if min_date <= datetime.date(year, month, calendar.monthrange(year, month)[1])
+                        and datetime.date(year, month, 1) <= max_date
+                    ]
+            period = sequential_ranges[period_type]
+
+            # # Which is less ugly?
+            # sequential_ranges['year']['range'] = xrange(min_date.year, max_date.year + 1)
+            # if period_type != 'year':
+            #     for year in sequential_ranges['year']['range']:
+            #         for month in easy_ranges['month_of_year']['range']:
+            #             days_in_month = calendar.monthrange(year, month)[1]
+            #             if (min_date <= datetime.date(year, month, days_in_month) and
+            #                     datetime.date(year, month, 1) <= max_date):
+            #                 sequential_ranges['month']['range'].append((year, month))
+            #             if period_type == 'day':
+            #                 for day in xrange(1, days_in_month + 1):
+            #                     if (min_date <= datetime.date(year, month, day) and
+            #                             datetime.date(year, month, day) <= max_date):
+            #                         sequential_ranges['day']['range'].append((year, month, day))
+            #             if period_type == 'week':
+            #                 for week in easy_ranges['week_of_year']['range']:
+            #                     if (min_date <= datetime.date(year, month, days_in_month) and
+            #                             datetime.date(year, month, 1) <= max_date):
+            #                         sequential_ranges['week']['range'].append((year, week))
+
+            period = sequential_ranges[period_type]
         else:
             raise ParseError(detail=('row_/col_period_type must be one of {}; received {}'
-                                     .format(easy_ranges.keys() + 'year', period_type)))
+                                     .format(easy_ranges.keys() + sequential_ranges.keys(),
+                                             period_type)))
 
         whens = []  # Eventual list of When-clause objects
-        when_lookup = 'occurred_from__{}'.format(period_type)
-        for x in period_range:
-            when_args = {'then': Value(x)}
-            when_args[when_lookup] = x
+        for x in period['range']:
+            when_args = period['lookup'](x)
+            when_args['then'] = Value(str(x))
             whens.append(When(**when_args))
 
-        labels = [{'key': str(x), 'label': period_labels(x)} for x in period_range]
-        return (Case(*whens, output_field=IntegerField()), labels)
+        labels = [{'key': str(x), 'label': period['label'](x)} for x in period['range']]
+        return (Case(*whens, output_field=CharField()), labels)
```

Comment:
The output_field is the key, and formerly those were all numbers, but now a bunch of them are tuples, so I'm casting them to strings (see also the change from `Value(x)` to `Value(str(x))` just above).